week1/utilities/data_prepper.py

```diff
@@ -232,20 +232,26 @@ def __log_ltr_query_features(self, query_id, key, query_doc_ids, click_prior_que
                                                 size=len(query_doc_ids), terms_field=terms_field)
         ##### Step Extract LTR Logged Features:
         # IMPLEMENT_START --
-        print("IMPLEMENT ME: __log_ltr_query_features: Extract log features out of the LTR:EXT response and place in a data frame")
+        # print("IMPLEMENT ME: __log_ltr_query_features: Extract log features out of the LTR:EXT response and place in a data frame")
+        response = self.opensearch.search(body=log_query, index=self.index_name)
+        # print(response)
         # Loop over the hits structure returned by running `log_query` and then extract out the features from the response per query_id and doc id.  Also capture and return all query/doc pairs that didn't return features
         # Your structure should look like the data frame below
         feature_results = {}
         feature_results["doc_id"] = []  # capture the doc id so we can join later
         feature_results["query_id"] = []  # ^^^
         feature_results["sku"] = []
         feature_results["name_match"] = []
-        rng = np.random.default_rng(12345)
-        for doc_id in query_doc_ids:
+        for doc in response['hits']['hits']:
+            doc_id = doc['_id']
             feature_results["doc_id"].append(doc_id)  # capture the doc id so we can join later
             feature_results["query_id"].append(query_id)
-            feature_results["sku"].append(doc_id)  
-            feature_results["name_match"].append(rng.random())
+            feature_results["sku"].append(doc_id)
+            name_match_value = doc['fields']['_ltrlog'][0]['log_entry'][0].get('value')
```

Comment:
This could perhaps have been made more generic to also capture other features you may add. I think also this current approach might hit problems when there is more than one feature present (ie the first on is not name_match but something else)